core/schema/git.go

```diff
@@ -191,3 +192,53 @@ func defaultBranch(ctx context.Context, repoURL string) (string, error) {
 
 	return "", fmt.Errorf("could not deduce default branch from output:\n%s", string(stdoutBytes))
 }
+
+// find all git tags for a given repo
+func gitTags(ctx context.Context, repoURL string) ([]string, error) {
+	stdoutBytes, err := exec.CommandContext(ctx, "git", "ls-remote", "--tags", "--symref", repoURL).Output()
```

Comment:
nit: we should include `--refs` here as well:

```
       --refs
           Do not show peeled tags or pseudorefs like HEAD in the output.
```

E.g. on the dagger repo, without this, we get the annotated tag `refs/tags/v0.6.1^{}`.

Reply:
This sounds reasonable to me.

Would you like to go ahead and commit this so that we can merge?

Reply:
Done.

Reply:
Good catch, I watched out for this but misremembered the `--symref` as the flag that handles that. :face_exhaling: